index.js

```diff
@@ -43,6 +43,144 @@
+//.     []                                  Nothing
+//.
+//.     |> R.tail (['foo'])                 |> S.tail (['foo'])
+//.     []                                  Just ([])
+//.
+//.     |> R.replace (/^x/) ('') ('abc')    |> S.stripPrefix ('x') ('abc')
+//.     'abc'                               Nothing
+//.
+//.     |> R.replace (/^x/) ('') ('xabc')   |> S.stripPrefix ('x') ('xabc')
+//.     'abc'                               Just ('abc')
+//.
+//. ### Invariants
+//.
+//. Sanctuary performs rigorous [type checking][] of inputs and outputs, and
+//. throws a descriptive error if a type error is encountered. This allows bugs
+//. to be caught and fixed early in the development cycle.
+//.
+//. Ramda operates on the [garbage in, garbage out][GIGO] principal. Functions
+//. are documented to take arguments of particular types, but these invariants
+//. are not enforced. The problem with this approach in a language as
+//. permissive as JavaScript is that there's no guarantee that garbage input
+//. will produce garbage output ([ramda/ramda#1413][]). Ramda performs ad hoc
+//. type checking in some such cases ([ramda/ramda#1419][]).
+//.
+//. Sanctuary can be configured to operate in garbage in, garbage out mode.
+//. Ramda cannot be configured to enforce its invariants.
+//.
+//. ### Currying
+//.
+//. Sanctuary functions are curried. There is, for example, exactly one way to
+//. apply `S.reduce` to `S.add`, `0`, and `xs`:
+//.
+//.   - `S.reduce (S.add) (0) (xs)`
+//.
+//. Ramda functions are also curried, but in a complex manner. There are four
+//. ways to apply `R.reduce` to `R.add`, `0`, and `xs`:
+//.
+//.   - `R.reduce (R.add) (0) (xs)`
+//.   - `R.reduce (R.add) (0, xs)`
+//.   - `R.reduce (R.add, 0) (xs)`
+//.   - `R.reduce (R.add, 0, xs)`
+//.
+//. Ramda supports all these forms because curried functions enable partial
+//. application, one of the library's tenets, but `f(x)(y)(z)` is considered
+//. too unfamiliar and too unattractive to appeal to JavaScript programmers.
+//.
+//. Sanctuary's developers prefer a simple, unfamiliar construct to a complex,
+//. familiar one. Familiarity can be acquired; complexity is intrinsic.
+//.
+//. The lack of breathing room in `f(x)(y)(z)` impairs readability. The simple
+//. solution to this problem, proposed in [#438][], is to include a space when
+//. applying a function: `f (x) (y) (z)`.
+//.
+//. Ramda also provides a special placeholder value, [`R.__`][], which removes
+//. the restriction that a function must be applied to its arguments in order.
+//. The following expressions are equivalent:
+//.
+//.   - `R.reduce (R.__, 0, xs) (R.add)`
+//.   - `R.reduce (R.add, R.__, xs) (0)`
+//.   - `R.reduce (R.__, 0) (R.add) (xs)`
+//.   - `R.reduce (R.__, 0) (R.add, xs)`
+//.   - `R.reduce (R.__, R.__, xs) (R.add) (0)`
+//.   - `R.reduce (R.__, R.__, xs) (R.add, 0)`
+//.
+//. ### Variadic functions
+//.
+//. Ramda provides several functions with take any number of arguments. These
```

Comment:
Typo `with` -> `which`

Reply:
I must have read this sentence a dozen times without noticing that. Well spotted, @jazen. Would you like to open a pull request to receive credit for having sharp eyes?

Reply:
Sure, thanks 😄